main.go

```diff
@@ -931,211 +924,93 @@ func getAlleles(chrom string, pos string, ref string, alt string) (string, []str
 	return parse.Snp, positions, references, alleles, indexes
 }
 
-// Current limitations: Does not support alleleIdx > 9, or sites which have 2 digits allele numbers
-// This allows us to improve performance, decrease code verbosity
-// Most multialellics with > 10 alleles will be false positives
-// because a true site would require a mutation rate of >> 1e-8 (say in chromosomal instability) and 10k samples
-// or an effective population size of billions (vs 10k expected) ; else .001^10 == 10^-30 == never happens
-
-// TODO: decide whether to be more strict about missing genotypes
-// Currently some garbage like .... would be considered "missing"
-func makeHetHomozygotes(fields []string, header []string, alleleNum rune) ([]string, []string, []string, int, int) {
-	simpleGt := !strings.Contains(fields[formatIdx], ":")
-
+// makeHetHomozygotes process all sample genotype fields, and for a single alleleNum, which is the allele index (1 based)
+// returns the homozygotes, heterozygotes, missing samples, total alt counts, genotype counts, and missing counts
+func makeHetHomozygotes(fields []string, header []string, alleleNum string) ([]string, []string, []string, int, int) {
 	var homs []string
 	var hets []string
 	var missing []string
 
-	var gt []string
-
 	var gtCount int
 	var altCount int
 
 	var totalAltCount int
 	var totalGtCount int
 
-	// Unfortunately there is no guarantee that genotypes will be consistently phased or unphased
-	/*  From https://samtools.github.io/hts-specs/VCFv4.1.pdf
-	#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
-	20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.
-	20 17330 . T A 3 q10 NS=3;DP=11;AF=0.017 GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 0/0:41:3
-	20 1110696 rs6040355 A G,T 67 PASS NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 2/2:35:4
-	20 1230237 . T . 47 PASS NS=3;DP=13;AA=T GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2
-	20 1234567 microsat1 GTC G,GTCT 50 PASS NS=3;DP=9;AA=G GT:GQ:DP 0/1:35:4 0/2:17:2 1/1:40:3
-	*/
 SAMPLES:
 	// NOTE: If any errors encountered, all genotypes in row will be skipped and logged, since
 	// this represents a likely corruption of data
-	for i := 9; i < len(header); i++ {
-		// If any 1 allele missing the genotype is, by definition missing
-		// Applies to haploid as well as diploid+ sites
-		if fields[i][0] == '.' {
-			missing = append(missing, header[i])
-			continue
-		}
-
-		// haploid
-		if len(fields[i]) == 1 || fields[i][1] == ':' {
-			if fields[i][0] == '0' {
-				totalGtCount++
-				continue
-			}
-
-			// We don't support haploid genotypes very well; I will count such sites
```

Comment:
Is the new version going to represent haploid genotypes as haploid or still homozygous?

Reply:
It will still be homozygous; in the for loop, each genotype value adds 1 count to the genotype count; if that genotype value matches the alleleNum (the multi allelic allele in the decomposed multi allelic, where alleleNum will be the only ALT allele on that line in the output of bystro-vcf), then alt also gets a count of 1.
- https://github.com/bystrogenomics/bystro-vcf/blob/dfb90652861ff50b24dc2e6e9099138185792c20/main.go#L1017

In the haploid case, if the only genotype value matches the alleleNum, that site will be homozygous/hemizygous, since `altCount == 1` and `gtCount == 1`